minimalkv/__init__.py

```diff
@@ -417,50 +378,87 @@ def _valid_ttl(self, ttl_secs):
+    ):
+        """Store bytestring data at key.
 
-        :param ttl_secs: Number of seconds until the key expires. See above
-                         for valid values.
-        :raises ValueError: If ``ttl_secs`` is invalid.
-        :raises IOError: If storing failed or the file could not
-                         be read
+        If *ttl_secs* is a positive number, the key will expire after *ttl_secs*. Other
+        possible values for *ttl_secs* are `minimalkv.FOREVER` (no expiration) and
+        `minimalkv.NOT_SET` (no TTL configuration). `None` will be replaced with
+        `minimalkv.TimeToLiveMixin.default_ttl_secs`.
 
+        :param key: The key under which the data is to be stored.
+        :param data: Data to be stored at key, must be `bytes`.
+        :param ttl_secs: Number of seconds until the key expires.
+
+        :returns: The key under which data was stored.
+
+        :raises ValueError: If the key is not valid.
+        :raises IOError: If storing failed or the file could not be read.
+        :raises ValueError: If ``ttl_secs`` is invalid.
         """
-        self._check_valid_key(key)  # type: ignore
+        self._check_valid_key(key)
         if not isinstance(data, bytes):
             raise IOError("Provided data is not of type bytes")
         return self._put(key, data, self._valid_ttl(ttl_secs))
 
-    def put_file(self, key: str, file, ttl_secs=None):
-        """Like :meth:`~minimalkv.KeyValueStore.put_file`, but with an
-        additional parameter:
+    def put_file(
+        self,
+        key: str,
+        file: Union[str, File],
+        ttl_secs: Optional[Union[float, int, str]] = None,
+    ):
+        """Store contents of file at key.
+
+        Store data from a file into key. *file* can be a string, which will be
+        interpreted as a filename, or an object with a *read()* method.
+
+        If *file* is a filename, the file might be removed while storing to avoid
+        unnecessary copies. To prevent this, pass the opened file instead.
+
+        If *ttl_secs* is a positive number, the key will expire after *ttl_secs*. Other
+        possible values for *ttl_secs* are `minimalkv.FOREVER` (no expiration) and
+        `minimalkv.NOT_SET` (no TTL configuration). `None` will be replaced with
+        `minimalkv.TimeToLiveMixin.default_ttl_secs`.
+
+        :param key: Key where to store data in file.
+        :param file: A filename or an object with a read method.
+        :param ttl_secs: Number of seconds until the key expires.
 
-        :param ttl_secs: Number of seconds until the key expires. See above
-                         for valid values.
+        :returns: The key under which data was stored.
+
+        :raises ValueError: If the key is not valid.
+        :raises IOError: If there was a problem moving the file in.
         :raises ValueError: If ``ttl_secs`` is invalid.
         """
+        # TODO: Remove next two lines?
```

Comment:
Why?

Reply:
`None` get's replaced with `self.default_ttl_secs` in `self.valid_ttl`.